api/api.go

```diff
@@ -12,6 +14,10 @@ import (
 	"github.com/julienschmidt/httprouter"
 )
 
+var (
+	errCleanClosed = errors.New("Client closed connection")
```

Comment:
I would probably prefer `errResponseWriterClosed`, and then also include one or two words in the error message to indicate that `Close` was called on the response writer prior to whatever call just failed. Object-specific error messages are useful, otherwise you can end up in situations where your error is something like `EOF` and you realize that there are probably 15 different places in your codepath that could have returned `EOF`.